### DISCUSSION TITLE
# Issue with Time Segment in `powerline-icon` Theme in Oh-My-Bash

### DISCUSSION THREAD

**Discussion by AbrarUS** [ohmybash/oh-my-bash#637] [Q&A] [answered]
Hi everyone,

I am using the `powerline-icon` theme in Oh-My-Bash and have been trying to customize the terminal prompt to include a time segment and change the colors of segments. Here's what I have achieved so far:

### Customizations I Made

1. **Changing Colors:**  
   I edited `CLOCK_THEME_PROMPT_COLOR=1` and `CWD_THEME_PROMPT_COLOR=30` (etc.) in the `~/.oh-my-bash/themes/powerline-icon/powerline-icon.theme.sh` file. This successfully changed the colors of the segments.

2. **Adding Time Segment:**  
   - I updated `.bashrc` with:  
     ```bash
     POWERLINE_PROMPT="user_info clock battery scm python_venv ruby cwd"
     ```
   - This enabled the clock segment in the terminal prompt.  
   - Since my coding environment is in a CS50.dev codespace (hosted on a remote server), I adjusted the time format to match my local time using the following in `powerline-icon.theme.sh`:  
     ```bash
     THEME_CLOCK_FORMAT=$(date -d "+5 hours 30 minutes" +"%H:%M:%S")
     THEME_CLOCK_FORMAT=${THEME_CLOCK_FORMAT:="%H:%M:%S"}
     ```

### Issue Faced
- The time displays correctly but does not update dynamically with each command. I have to run `source ~/.bashrc` to see the updated time (example below).  
  ![Static time issue](https://github.com/user-attachments/assets/a9770e72-5943-429e-bebe-d419a70b5c81)

- If I remove the custom `THEME_CLOCK_FORMAT` line, the time updates dynamically but is incorrect due to the remote server's timezone. For instance, it shows 12:17 instead of 17:47.  
  ![Incorrect time issue](https://github.com/user-attachments/assets/5421eafa-a410-4632-aa28-5458e0c04178)

### Attempts to Solve
- Adding `clock` in the `POWERLINE_PROMPT` directly within `powerline-icon.theme.sh`:
  ```bash
  POWERLINE_PROMPT=${POWERLINE_PROMPT:="user_info clock battery scm python_venv ruby cwd"}
  ```
  This did not resolve the issue.

- Experimenting with `OMB_THEME_POWERLINE_ICON_CLOCK` as per the `README.md`:
  ```bash
  : "${OMB_THEME_POWERLINE_ICON_CLOCK=%X %D}"
  ```
  But this also had no effect.  
  ![Experiment with OMB_THEME_POWERLINE_ICON_CLOCK](https://github.com/user-attachments/assets/8071ff83-4399-41cc-968f-5fe6bd1083cd)

### Request for Help
I’m new to terminal themes and commands, and this is the result of hours of trial and error. I’d greatly appreciate any guidance or suggestions to:

1. Ensure the time segment updates dynamically while showing the correct local time.
2. Provide more clarity on the usage of `OMB_THEME_POWERLINE_ICON_CLOCK` or other related configurations.
3. Share resources for better understanding Oh-My-Bash theme customization.

Thank you in advance for your help!

**Comment by akinomyoga** [comment, marked as answer]
> ```bash
> THEME_CLOCK_FORMAT=$(date -d "+5 hours 30 minutes" +"%H:%M:%S")
> ```

This sets `THEME_CLOCK_FORMAT` to a static string that contains the time at which Oh My Bash is loaded (i.e., the shell startup time).

Instead of trying to change the format, have you tried setting the environment variable `TZ`?

**Reply by AbrarUS** [reply]
Just did it now. It works! 😊 
Added `export TZ='Asia/Kolkata'` to the bashrc and removed `THEME_CLOCK_FORMAT=$(date -d "+5 hours 30 minutes" +"%H:%M:%S")` from powerline-icon.theme.sh  

This works well. Now I want to know how do I learn of all these? Are there good videos, resources, etc.

**Reply by akinomyoga** [reply]
You first need to be familiar with the (POSIX) shell language. Then, you also want to learn basic command-line utilities. One of the most basic documents is [the CLI section of the POSIX standard](https://pubs.opengroup.org/onlinepubs/9799919799/idx/xcu.html). You may also want to learn the Bash language, which is an extension of the POSIX shell language.
